workers/store-config/src/objects/OrderIdObject.ts

```diff
@@ -7,53 +7,54 @@ export class OrderIdObject {
         this.env = env;
     }
 
+    // Asynchronously hashes an input string via SHA-256.
+    private async hash(message: string): Promise<string> {
+        const data = new TextEncoder().encode(message);
+        const hashBuffer = await crypto.subtle.digest('SHA-256', data);
+        const hashArray = Array.from(new Uint8Array(hashBuffer));
+        const hashHex = hashArray
+            .map((byte) => byte.toString(16))
+            .join('')
+            .toUpperCase();
+        return hashHex;
+    }
+
+    // NB: this will not work if there are >= 1e3 requests on the same exact ms.
+    // Atomically returns and increments the unique counter stored in DO.
+    private async sequence() {
+        let val;
+        await this.state.storage.transaction(async (txn) => {
+            val = await txn.get('counter');
+            val = val === undefined || val === 1e3 ? 0 : val;
+            await txn.put('counter', val + 1);
+        });
+        return val;
+    }
+
     // Handle HTTP requests from clients.
-    async fetch(    
-        request: Request
-    ) {
+    async fetch(request: Request) {
         const HT_PREFIX = this.env.NODE_ENV === 'PROD' ? 'HT' : 'HTT';
-        
-        // Current date, YYMMDD => 230429 - This will break in year 2100 lol
-        const date = new Date()
-            .toISOString()
-            .split('T')[0]
-            .replaceAll('-', '')
-            .slice(2);
-
-        // Fetch exisiting sequence from DO
-        const storedValue = await this.state.storage?.get('value') as string;
-
-        // Make next sequence number part, 3 digits
-        // TODO what if we exceed 999, for now we will just go directly to 1000
-        let nextSeq;
-        if (
-            !storedValue ||
-            storedValue.slice(HT_PREFIX.length, HT_PREFIX.length + 6) !== date
-        ) {
-            nextSeq = getRndInteger(2, 10);
-        } else {
-            nextSeq = parseInt(storedValue.slice(HT_PREFIX.length + 6)) + 1;
-        }
-        nextSeq = nextSeq.toString().padStart(3, '0');
+
+        // Atomically generate and put the unique identifier
+        const nextSeq = await this.sequence();
```

Comment:
[On second thought] Perhaps we should couple the padding length as the sequence reset length - let me know what you think

Reply:
Yes, either this or define the reset length at the class level. As long as its one place where 1e3 is defined